src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SetDefaultResponseHeadersMiddleware.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+/// <summary>
+/// System.Web would set some response headers by default. This adds those early in the pipeline so they can still be overridden. If they exist, they will not be replaced.
+/// </summary>
+internal class SetDefaultResponseHeadersMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    private readonly (string Header, string Value)[] _defaultHeaders = new (string, string)[]
+    {
+        (HeaderNames.CacheControl, CacheControlHeaderValue.PrivateString),
+        (HeaderNames.ContentType, "text/html")
+    };
+
+    public SetDefaultResponseHeadersMiddleware(RequestDelegate next) => _next = next;
+
+    public Task InvokeAsync(HttpContext context)
+    {
+        foreach (var (header, value) in _defaultHeaders)
+        {
+            if (!context.Response.Headers.ContainsKey(header))
```

Comment:
Since we only set headers that aren't already present, should we call `_next` first so that whatever endpoint handles the request has an opportunity to write headers and we'll only set the default ones if they're still unset after that?

Reply:
We can't really do that because after `_next(context)`  the response will have started writing. The alternative would be to register a  `Response.OnStarting` delegate that would handle it there. Let me think about the implications of either.

Reply:
That's a good point. I'm ok with keeping this model if it's simpler.